### DISCUSSION TITLE
Handle Prometheus API without /api/v1/status/config endpoint (e.g. Thanos)

### DISCUSSION THREAD

**Discussion by filippog** [cloudflare/pint#928] [General]
This is a followup issue from #573.

Currently `pint` will fail checks that require Prometheus configuration to work e.g. `promql/rate`, for example when running `pint` against a Thanos cluster.

There's several approaches to this problem, each with their own trade offs, including:
1. warn the user something is amiss
1. disable the checks that require access to Prometheus configuration
1. synthesize the required Prometheus configuration in `pint` configuration

**Comment by zerodayyy** [comment]
I've got a similar issue when running Pint against VictoriaMetrics. It doesn't provide some Prometheus API endpoints like Thanos, leading to similar errors.

**Comment by hagen1778** [comment]
@zerodayyy if pint uses `/api/v1/status/config` call only to get `scrape_interval` value for `rate` validation, then it makes not much sense for VictoriaMetrics due to the following reasons:
1. VictoriaMetrics doesn't need to correct too low look-behind window for `rate` expressions, as it does it automatically during evaluation. See https://www.percona.com/blog/better-prometheus-rate-function-with-victoriametrics/ and https://docs.victoriametrics.com/metricsql
2. VictoriaMetrics is usually used as centralized storage for many collectors. The data can be scraped with different scrape_intervals, the data can be pushed with custom resolution. So it could be that global `scrape_interval` just doesn't make sense, since various metrics could have different resolution.

Btw, Prometheus can also have jobs with custom scrape intervals. Does `pint` account for this by checking to which job metrics selector in the expression belongs?

**Comment by jlaberge-league** [comment]
Also running into a similar issue due to these two endpoints. My use case is Google Managed Prometheus which does not expose those APIs from Prometheus. Simply disabling some checks (while not ideal either) does not get rid of the frequent calls to `/api/v1/status/config` as it's also being used by the watch command.

It seems any issue opened up regarding these endpoints is converted into a conversation, so I'm unsure if this is something we can expect to see support for?

Sorry for the direct tag @prymitive , but it seems as though you are the primary contributor so I'd be interested to hear your thoughts.
